src/Quantum/MatrixProductState.jl

```diff
@@ -185,105 +185,93 @@ function Base.rand(rng::Random.AbstractRNG, sampler::MPSSampler{Closed,T}) where
     return ψ
 end
 
-# TODO modify ψ in place
+function truncatedims(t::Tensor, dim::Union{Integer, Symbol}, end_ind::Integer)
+    if dim isa Symbol
+        dim = findfirst(==(dim), labels(t))
+    end
+
+    data = view(parent(t), ntuple(i -> i == dim ? (1:end_ind) : Colon(), ndims(t))...)
+    new_labels = labels(t)
+
+    return Tensor(data, new_labels; t.meta...)
+end
+
```

Comment:
Don't use this function. Use `@view` on the tensor.

Reply:
I am using `view` in the code